doc/api/worker_threads.md

```diff
@@ -61,6 +61,26 @@ Worker threads inherit non-process-specific options by default. Refer to
 [`Worker constructor options`][] to know how to customize worker thread options,
 specifically `argv` and `execArgv` options.
 
+## `worker.connect(target[, data][, timeout])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active development
+
+* `target` {number} The target thread id.
+* `data` {any} Any arbitrary, cloneable JavaScript value.
+* `timeout` {number} Time to wait in milliseconds. By default it's `undefined`, which
+  means wait forever.
+* Returns: {Promise} A promise for a `MessagePort`.
```

Comment:
Does this really need to be a `Promise<MessagePort>`? It would be nice if it could return a `MessagePort` directly and just have it so that any messages posted are held until any underlying setup is complete?

Reply:
At the beginning it was actually sync, as I was only using `Atomics.wait`. But then I discovered this would cause deadlocks since the thread will not be able to accept connection from other threads while waiting for a reply.
I move all to be async.

Anyway, in the future we can add a sync version if needed.

Reply:
I don't mean block until the connection is established. I mean return a `MessagePort` immediately but still perf the underlying connection asynchronously and arrange for any messages posted to the `MessagePort` during that time to be queued until the underlying connection is established.

Reply:
I honestly think the queueing might be too much.
Most of the times I expect this API to be mostly sync-like and return right away.
Are we good in eventually add a `connectSync` API which works like that in a future iteration?